packages/airnode-deployer/src/handlers/aws/index.ts

```diff
@@ -44,7 +44,10 @@ export async function run(payload: WorkerPayload): Promise<AWSLambda.APIGatewayP
 // https://api3dao.atlassian.net/browse/AN-527
 
 async function initializeProvider(payload: InitializeProviderPayload) {
-  const stateWithConfig = { ...payload.state, config: parsedConfig };
+  const { state } = payload;
+  const { chainId, name: providerName } = state.settings;
+  addMetadata({ 'Chain-ID': chainId, Provider: providerName });
```

Comment:
These could also be set in the `run` function but I guess it would have to be with some conditional checks because they are different for each of the handlers, so probably best to leave like this :thinking:

Reply:
Yeah, we could have the chainID set right away in the run handler, but I do not see much point in doing so since there need to be the individual `addMetadata` calls.